draft-ietf-quic-qpack.md

```diff
@@ -52,46 +52,162 @@ code and issues list for this draft can be found at
+improve stream multiplexing relative to HTTP/2 by reducing head-of-line
+blocking.  If HPACK were used for HTTP/QUIC, it would induce head-of-line
+blocking due to built-in assumptions of a total ordering across frames on all
+streams.
 
 QUIC is described in {{?QUIC-TRANSPORT=I-D.ietf-quic-transport}}.  The HTTP/QUIC
 mapping is described in {{!QUIC-HTTP=I-D.ietf-quic-http}}. For a full
 description of HTTP/2, see {{?RFC7540}}. The description of HPACK is
 {{!RFC7541}}, with important terminology in Section 1.3.
 
-QPACK modifies HPACK to allow correctness in the presence of out-of-order
-delivery, with flexibility for implementations to balance between resilience
-against HoL blocking and optimal compression ratio.  The design goals are to
-closely approach the compression ratio of HPACK with substantially less
-head-of-line blocking under the same loss conditions.
+QPACK reuses core concepts from HPACK, but is redesigned to allow correctness in
+the presence of out-of-order delivery, with flexibility for implementations to
+balance between resilience against head-of-line blocking and optimal compression
+ratio.  The design goals are to closely approach the compression ratio of HPACK
+with substantially less head-of-line blocking under the same loss conditions.
 
-QPACK is intended to be a relatively non-intrusive extension to HPACK; an
-implementation should be easily shared within stacks supporting both HTTP/2 over
-(TLS+)TCP and HTTP/QUIC.
+## Header Tables
 
-## Head-of-Line Blocking in HPACK {#hol-example}
+Like HPACK, QPACK uses two tables for associating header fields to indexes.  The
+static table (see {{table-static}}) is predefined and contains common header
+fields (some of them with an empty value).  The dynamic table (see
+{{table-dynamic}}) built up over the course of the connection and can be used by
+the encoder to index header fields repeated in the encoded header lists.
 
-HPACK enables several types of header representations, one of which also adds
-the header to a dynamic table of header values.  These values are then available
-for reuse in subsequent header blocks simply by referencing the entry number in
-the table.
+These two tables are indexed in a different manner than in HPACK; see
+{{indexing}} for more details.
 
-If the packet containing a header is lost, that stream cannot complete header
-processing until the packet is retransmitted.  This is unavoidable. However,
-other streams which rely on the state created by that packet *also* cannot make
-progress. This is the problem which QUIC solves in general, but which is
-reintroduced by HPACK when the loss includes a HEADERS frame.
+### Static Table {#table-static}
+
+The static table consists of a predefined static list of header fields.  Its
+entries are defined in Appendix A of {{!RFC7541}}. Note that because HPACK did
+not use zero-based references, there is no value at index zero of the static
+table.
+
+### Dynamic Table {#table-dynamic}
```

Comment:
This would benefit from subsections so that people can find Absolute and Relative indexing, "Base Index", Post-Base, etc...